vyxal/helpers.py

```diff
@@ -770,14 +770,34 @@ def transfer_capitalisation(source: str, target: str) -> str:
     return ret
 
 
+@lazylist
 def transpose(
-    vector: VyList, filler: Any = None, ctx: Context = DEFAULT_CTX
+    matrix: VyList, filler: Optional[Any] = None, ctx: Context = DEFAULT_CTX
 ) -> VyList:
-    """Transposes a vector"""
-    vector = iterable(vector, ctx=ctx)
-    temp = itertools.zip_longest(*map(iterable, vector), fillvalue=filler)
+    """Transposes a matrix
+    In order to handle infinite lists, it generates the transpose
+    antidiagonal by antidiagonal.
+    """
+    matrix = iterable(matrix, ctx=ctx)
+    matrix = vy_map(iterable, matrix, ctx=ctx)
 
-    return vyxalify((item for item in x if item is not None) for x in temp)
+    @lazylist
+    def gen_row(r: int):
+        c = 0
+        while has_ind(matrix, c):
+            if has_ind(matrix[c], r):
+                yield matrix[c][r]
+            elif filler is not None:
+                yield filler
+            c += 1
+
+    r = 0
+    while r < 10:
```

Comment:
What exactly does this do?

Reply:
Oh shoot, that's debugging code, it should be `while True:`